### PR TITLE
8309935: Mac - SystemMenuBar, IndexOutOfBoundsException on change

### PR DESCRIPTION
**Pull request by FlorianKirmaier**
Fixing JDK-8309935 and providing a test for it.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8309935](https://bugs.openjdk.org/browse/JDK-8309935): Mac - SystemMenuBar, IndexOutOfBoundsException on change (**Bug** - P4)

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1152/head:pull/1152` \
`$ git checkout pull/1152`

Update a local copy of the PR: \
`$ git checkout pull/1152` \
`$ git pull https://git.openjdk.org/jfx.git pull/1152/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1152`

View PR using the GUI difftool: \
`$ git pr show -t 1152`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1152.diff">https://git.openjdk.org/jfx/pull/1152.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1152#issuecomment-1589375623)

### CODE REVIEW

File: tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
+package test.javafx.stage;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+import test.util.memory.JMemoryBuddy;
```
Comment (kevinrushforth) [inline tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java:7]: Unused import.

**Review by kevinrushforth — COMMENTED** [review]
The fix and the test look OK to me. Have you tested it on all platforms (including Linux and Windows, which will ignore the attempt to use a system menu bar, but should otherwise behave as expected)?

I left a couple inline comments.

**Comment by FlorianKirmaier** [comment]
I've tested it on Mac.

On Windows, many people will test it soon, because the fix is now integrated into a development version of a JavaFX Application, with a lot of Testers / Developers. So if something breaks, we should know it very soon. 
But I don't expect any issues.

For Linux, I currently don't have a device available to test it.
I could create a TestVM, to just Test it, but I don't really see how this breaks only on Linux.

File: tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package test.javafx.stage;
```
[the remark this answers is not in the document]
Comment (FlorianKirmaier) [reply tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java:26]: done

File: tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java (hunk 1 above)
Comment (FlorianKirmaier) [reply tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java]: done

**Review by kevinrushforth — APPROVED** [review]
Looks good. Tested with the provided system test on Mac, where it fails without and passes with the fix.

**Comment by kevinrushforth** [comment]
@FlorianKirmaier This is ready for you to `/integrate`

**Pull request closed without merging**